### PR TITLE
Add in_cluster_url parameters to kiali configmap

### PR DESCRIPTION
**Pull request by voroniys**
[ x] Configuration Infrastructure
[ ] Docs
[ x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

These options are the only way to get jaeger/grafana integration working for some kubernetes installations on AWS. Was initially proposed in #20057 for helmchart, but the pull request were closed without merging as the helm way of installation declared obsoleted. Therefore the new request for istioctl way.

### CODE REVIEW

File: manifests/istio-telemetry/kiali/templates/configmap.yaml [hunk 1]
```diff
@@ -34,8 +34,14 @@ data:
         url_service_version: http://istio-pilot.{{ .Values.global.configNamespace }}:8080/version
       tracing:
         url: {{ .Values.kiali.dashboard.jaegerURL }}
+{{- if .Values.kiali.dashboard.inclusterjaegerURL }}
+        in_cluster_url: {{ .Values.dashboard.inclusterjaegerURL }}
```
Comment (sdake) [inline manifests/istio-telemetry/kiali/templates/configmap.yaml]: should this not be `.Values.kiali.dashboard.inclusterjaegerURL`?

File: manifests/istio-telemetry/kiali/templates/configmap.yaml [hunk 2]
```diff
@@ -34,8 +34,14 @@ data:
...
+{{- end }}
       grafana:
         url: {{ .Values.kiali.dashboard.grafanaURL }}
+{{- if .Values.kiali.dashboard.inclustergrafanaURL }}
+        in_cluster_url: {{ .Values.dashboard.inclustergrafanaURL }}
```
Comment (sdake) [inline manifests/istio-telemetry/kiali/templates/configmap.yaml]: should this not be `.Values.kiali.dashboard.inclustergrafanaURL`?

**Review by sdake — COMMENTED** [review]
looks pretty good. Few questions inline. Please take care to add these values to values.yaml:

https://github.com/istio/istio/blob/master/manifests/istio-telemetry/kiali/values.yaml#L74-L75

**Comment by sdake** [comment]
@voroniys Welcome to the Istio community. We are in the final days of 1.5 development. If you could quickly (1-3 days) turn around the suggested changes or why the changes should not be made, we can get this merged.

Cheers
-steve

**Comment by sdake** [comment]
@voroniys the linter (lint_istio) is failing because valuess.yaml is not specified as determined by:
```
==> Linting manifests/istio-telemetry/kiali
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "kiali/templates/configmap.yaml": template: kiali/templates/configmap.yaml:37:14: executing "kiali/templates/configmap.yaml" at <.Values.kiali.dashboard.inclusterjaegerURL>: map has no entry for key "inclusterjaegerURL"
```

When changing the installation, please run `make gen` and commit the changes. There is significant auto-generated code, and as a project, we store the auto-generated code, rather than generating at build time.

File: manifests/istio-telemetry/kiali/templates/configmap.yaml (hunk 1 above)
Comment (voroniys) [reply manifests/istio-telemetry/kiali/templates/configmap.yaml]: You are right. Cut&paste issue :(

**Comment by sdake** [comment]
> @voroniys: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> gencheck_istio	[7eae423](https://github.com/istio/istio/commit/7eae423ad07aeb55d37981dce97e1e1a8551067a)	[link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/21393/gencheck_istio/7273)	`/test gencheck_istio`

When changing the installation, please run make gen and commit the changes. There is significant auto-generated code, and as a project, we store the auto-generated code, rather than generating at build time.

Cheers
-steve

**Comment by voroniys** [comment]
> 
> When changing the installation, please run `make gen` and commit the changes. There is significant auto-generated code, and as a project, we store the auto-generated code, rather than generating at build time.

After commiting `make gen` changes even more tests are failing :( All these generated files have now merge conflicts.

**Comment by voroniys** [comment]
As far as I can judge the failing unit test has nothing to do with my change.

**Review by sdake — APPROVED** [review]
@voroniys really nice work. I will do what I can to get this work backported to 1.5.

**Comment by sdake** [comment]
> As far as I can judge the failing unit test has nothing to do with my change.

Some of our tests don't correctly operate 100% of the time. It is frustrating for many folks. The best workaround we have at the current time is to follow the bot's instructions with a `/test what failed`. I've done that for you here, but you should be able to do this on your own.

Cheers
-steve

**Comment by jotak** [comment]
I did a similar one, that has already been merged in 1.5 but not in master (see https://github.com/istio/istio/pull/21255 ).
I fine if your PR supersedes mine. But take care of conflicts in the one that was merged (1.5)

**Comment by jotak** [comment]
(cc @dgn )

**Comment by jotak** [comment]
By the way you didn't set a default in `operator/data/profiles/default.yaml` . A good default would be `http://tracing.istio-system/jaeger`. Is it possible to add that?

**Comment by voroniys** [comment]
> By the way you didn't set a default in `operator/data/profiles/default.yaml` . A good default would be `http://tracing.istio-system/jaeger` for tracing, and similar for grafana. Is it possible to add that?

I framed this setting in configmap with if conditions instead. They are only added if they are set in user provided configuration, not by default.

**Comment by jotak** [comment]
@voroniys I think we should have a common solution between 1.5 and master, to avoid unnecessary config change (for helm chart) in between.

I think having a good default that works in default / demo istio install would be good, so that kiali can connect with jaeger out of the box (without specific config). "http://tracing.istio-system/jaeger" should work most of the time. Perhaps just "http://tracing/jaeger" (without namespace) would even be better. And same for Grafana.

Or do you see something wrong in having defaults?

**Comment by voroniys** [comment]
The default profile does not have kiali/jaeger/grafana installed by default.

**Comment by jotak** [comment]
@voroniys but the demo one does.

**Comment by sdake** [comment]
@jotak the typical process for merging work is to merge to master first then back port via our backport robot. The robot doesn’t always work. When it doesn’t, a manual cherry-pick should be done. I would suggest the following actions:

- Create a small surgical PR for master which addresses your concern and goes through the review process.
- Backport this PR.
- Backport the additional PR that changes default behavior.

We can review handling of defaults in the follow up PR as this PR has already merged.

**Comment by sdake** [comment]
@voroniys Thank you for your work here. Nice job in getting this work into master!

**Pull request merged**